src/Telemetry/AzureDevOpsTimeline/AzureDevOpsTimeline.cs

```diff
@@ -319,5 +353,49 @@ private static string GetBucket(AugmentedTimelineIssue augIssue)
         {
             return await azureServer.ListBuilds(project, cancellationToken, minDateTime, limit);
         }
+
+        private async Task GetImageNames(List<AugmentedTimelineRecord> records, CancellationToken cancellationToken)
+        {
+            SemaphoreSlim throttleSemaphore = new SemaphoreSlim(50);
+
+            var taskList = new List<Task>();
+            int successfulTasksCount = 0;
+
+            foreach (var record in records)
+            {
+                if (!string.IsNullOrEmpty(record.Raw.Log?.Url) && record.Raw.Name == "Initialize job")
+                {
+                    var childTask = Task.Run(async () =>
+                    {
+                        await throttleSemaphore.WaitAsync();
+
+                        try
+                        {
+                            if (!cancellationToken.IsCancellationRequested)
+                            {
+                                record.ImageName = (record.Raw.WorkerName.StartsWith("Azure Pipelines") || record.Raw.WorkerName.StartsWith("Hosted Agent"))
```

Comment:
This line could probably be broken up in an "if/elseif" set.  Its' a bit of a mouthful right now... a nested ternary expression with a relatively complicated condition.

Reply:
done